### PR TITLE
Improve `job describe` output

### PR DESCRIPTION
**Pull request by wdbaruni**
This PR improves `GET /api/v1/orchestrator/jobs/:id` API and `bacalhau job describe :id` CLI by including execution information in the output, and printing a more human readable format instead on just spitting out the job spec in YAML.

### API Improvements
`GET /api/v1/orchestrator/jobs/:id` now accepts an optional `?include="executions,history"` parameters where callers can ask for the job history and executions to be included in the response, instead of having to do separate calls to `GET /api/v1/orchestrator/jobs/:id/executions` and `GET /api/v1/orchestrator/jobs/:id/history` APIs to get the full picture of the job status. 

### CLI Improvements
`bacalhau job describe :id` now prints the results in different sections in a human readable format instead of the yaml spec as follows:

```
# run a job across three nodes where each print their time in nanos
bacalhau docker run --concurrency 3 ubuntu -- sh -c 'echo $(date +%s%N)'


# describe the job
→ bacalhau job describe e972a009-317e-45ec-b38e-3a40b5562141
ID            = e972a009-317e-45ec-b38e-3a40b5562141
Name          = e972a009-317e-45ec-b38e-3a40b5562141
Namespace     = 6af4ba5fa017d0b633e574e53fc773fee950f0d8bae9f7ff4e8f46d9e8e46cb1
Type          = batch
State         = Completed
Count         = 3
Created Time  = 2024-02-14 13:52:56
Modified Time = 2024-02-14 13:52:57
Version       = 1

Summary
Completed = 3

Executions
 ID          NODE ID  STATE      DESIRED  REV.  CREATED  MODIFIED  COMMENT
 e-0e2ff6bd  node-2   Completed  Stopped  6     7s ago   6s ago
 e-448d73d8  node-3   Completed  Stopped  6     7s ago   6s ago
 e-48cfe90d  node-1   Completed  Stopped  6     7s ago   6s ago

Standard Output
Execution e-48cfe90d:
1707918776915793876

Execution e-0e2ff6bd:
1707918776786937043

Execution e-448d73d8:
1707918776883765584

# Describe in yaml format
→ bacalhau job describe e972a009-317e-45ec-b38e-3a40b5562141 --output yaml
Executions:
  Executions:
  - AllocatedResources:
      Tasks: {}
    ComputeState:
      StateType: 7
    ...
Job:
  Constraints: []
  Count: 3
  CreateTime: 1707918776028710000
  ID: e972a009-317e-45ec-b38e-3a40b5562141
  Labels: {}
  Meta:
    bacalhau.org/client.id: 6af4ba5fa017d0b633e574e53fc773fee950f0d8bae9f7ff4e8f46d9e8e46cb1
    bacalhau.org/requester.id: node-0
  ModifyTime: 1707918777303266000
  Name: e972a009-317e-45ec-b38e-3a40b5562141
  Namespace: 6af4ba5fa017d0b633e574e53fc773fee950f0d8bae9f7ff4e8f46d9e8e46cb1
  Priority: 0
  Revision: 3
  State:
    StateType: Completed
  Tasks:
  - Engine:
      Params:
        Entrypoint: null
        EnvironmentVariables: []
        Image: ubuntu
        Parameters:
        - sh
        - -c
        - echo $(date +%s%N)
        WorkingDirectory: ""
      Type: docker
    Name: main
    Network:
      Type: None
    Publisher:
      Type: noop
    Resources: {}
    ResultPaths:
    - Name: outputs
      Path: /outputs
    Timeouts:
      ExecutionTimeout: 1800
  Type: batch
  Version: 1

```

Closes https://github.com/bacalhau-project/expanso-planning/issues/402

### CODE REVIEW

File: cmd/cli/job/describe.go [hunk 1]
```diff
@@ -60,13 +64,127 @@ func (o *DescribeOptions) run(cmd *cobra.Command, args []string) {
...
+	var executions []*models.Execution
+	if response.Executions != nil {
+		// TODO: #520 rename Executions.Executions to Executions.Items
+		executions = response.Executions.Executions
+	}
+
+	o.printHeaderData(cmd, job)
+	o.printExecutionsSummary(cmd, executions, jobID)
+	o.printExecutions(cmd, executions, jobID)
+	o.printOutputs(cmd, executions)
```
Comment (rossjones) [inline cmd/cli/job/describe.go:92]: Might it be best if these returned errors so we could decorate them with "failed to display job" and have the errors returned have more useful info?

Comment (wdbaruni) [reply cmd/cli/job/describe.go:92]: Each of these methods are calling `util.Fatal` internally with a decorated error message. `util.Fatal` will call `os.Exit` and the following print commands won't be called. I can change this if you thing it is a better approach

Comment (wdbaruni) [reply cmd/cli/job/describe.go]: Actually your suggestion is better. Done in https://github.com/bacalhau-project/bacalhau/pull/3459/commits/9aca5d0d3f3bc7b2909970bee3190ea33d15d3b1

**Review by aronchick — APPROVED** [review]
LGTM

I think the multiple outputs feels right, and the look and feel is off to a great start. Let's check it in and see what breaks (llol) and tweak the UI when we finish the other work

**Pull request merged**